assign2/index.js

```diff
@@ -0,0 +1,302 @@
+let getOption = JSON.parse(getLocal);
+
+const plusOptions = getOption.filter((option) => {
+  return option.isPlus == true;
+});
+
+const minusOptions = getOption.filter((option) => {
+  return option.isPlus != true;
+});
+
+function optionMaking(options) {
+  options.forEach((option) => {
+    const { category, isPlus } = option;
+    const modalOption = document.createElement("option");
+    modalOption.innerText = category;
+    modalOption.value = category;
+
+    isPlus
+      ? incomeSelect.appendChild(modalOption)
+      : expenditureSelect.appendChild(modalOption);
+  });
+}
+
+optionMaking(plusOptions);
+optionMaking(minusOptions);
+
+//모달 옵션
+function selectCheck() {
+  if (incomeCheck.checked) {
+    expenditureSelect.style.display = "none";
+    incomeSelect.style.display = "flex";
+    selectOption = document.querySelector("#income_select");
+  } else if (expenditureCheck.checked) {
+    expenditureSelect.style.display = "flex";
+    incomeSelect.style.display = "none";
+    selectOption = document.querySelector("#expenditure_select");
+  }
+}
+incomeCheck.addEventListener("change", () => selectCheck());
+expenditureCheck.addEventListener("change", () => selectCheck());
+
+//5.d
+const saveBtn = document.getElementById("save_button");
+const howMuchInput = document.getElementById("howmuch_input");
+const summaryInput = document.getElementById("summary_input");
+
+//심화 2.b
+howMuchInput.addEventListener("input", (e) => {
+  if (isNaN(e.target.value.charAt(0))) {
+    alert("숫자만 입력 가능합니다");
+    howMuchInput.value = null;
+    return;
+  } else {
+    let value = howMuchInput.value.replace(/,/g, "");
+    value = addThousnadsSeparator(value);
+    howMuchInput.value = value;
+    return;
+  }
+});
+
+//심화 5
+function addThousnadsSeparator(num) {
+  return num.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+}
+
+function addList() {
+  let selectedOption = selectOption.value;
+  let howMuchSummaried = parseInt(howMuchInput.value.replace(/,/g, ""), 10);
+  let summarySummarized = summaryInput.value;
+
+  let newList = [
+    {
+      title: selectedOption,
+      summary: summarySummarized,
+      money: howMuchSummaried,
+      getto: incomeCheck.checked ? true : false,
+    },
+  ];
+
+  let ADD_BALANCE = 0;
+  let MODAL_INMONEY = 0;
+  let MODAL_OUTMONEY = 0;
+  newList.forEach((data) => {
+    const { money, getto } = data;
+    ADD_BALANCE += getto ? money : -money;
+    let TOTAL_BALANCE = Number(ADD_BALANCE) + Number(INIT_BALANCE);
+    totalMoney.innerText = TOTAL_BALANCE.toLocaleString();
+
+    getto ? (MODAL_INMONEY += money) : (MODAL_OUTMONEY += money);
+    let TOTAL_INMONEY = Number(IN_MONEY) + Number(MODAL_INMONEY);
+    let TOTAL_OUTMONEY = Number(OUT_MONEY) + Number(MODAL_OUTMONEY);
+    plusMoney.innerText = TOTAL_INMONEY.toLocaleString();
+    minusMoney.innerText = TOTAL_OUTMONEY.toLocaleString();
+  });
+
+  //심화 2.a
+  if (howMuchSummaried.length == 0 || summarySummarized.length == 0) {
```

Comment:
```suggestion
  if (!howMuchSummaried || !summarySummarized) {
```
이렇게 더 간단하게 표현할 수도 있겠다!